src/components/common/form/button.js

```diff
@@ -0,0 +1,30 @@
+/* eslint-disable react/button-has-type */
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+
+class Button extends PureComponent {
+  render() {
+    const {
+      icon,
+      type,
+      caption,
+    } = this.props;
+    return (
+      <button type={type}>
+        {icon && <i className="material-icons">{icon}</i>}
+        <span className="btn-caption">{caption}</span>
+      </button>
+    );
+  }
+}
+
+Button.defaultProps = {
+  icon: null,
```

Comment:
You don't need to set the default icon prop as `null`. Leaving the icon prop undefined works just the same.

Reply:
Yup I totally agree with you. I added this because my eslint was complaining `[eslint] propType "icon" is not required, but has no corresponding defaultProp declaration.`. I can disable this and remove the defaultProps if you think its better. I just was just trying to avoid disabling many of the rules that were already defined in the eslint but I guess sometimes it's okay since we don't have to agree to all the rules.

Reply:
Ah ok. I didn't get this error when I ran eslint on this branch. It's safe to remove the defaultProp as it is redundant here.